gateway-js/src/index.ts

```diff
@@ -255,91 +288,94 @@ export class ApolloGateway implements GraphQLService {
-      const loglevelLogger = loglevel.getLogger(`apollo-gateway`);
-
-      // And also support the `debug` option, if it's truthy.
-      if (this.config.debug === true) {
-        loglevelLogger.setLevel(loglevelLogger.levels.DEBUG);
-      } else {
-        loglevelLogger.setLevel(loglevelLogger.levels.WARN);
-      }
-
-      this.logger = loglevelLogger;
+      return this.config.logger;
     }
 
-    if (isLocalConfig(this.config)) {
-      const { schema, composedSdl } = this.createSchema({
-        serviceList: this.config.localServiceList,
-      });
-      this.schema = schema;
+    // If the user didn't provide their own logger, we'll initialize one.
+    const loglevelLogger = loglevel.getLogger(`apollo-gateway`);
 
-      if (!composedSdl) {
-        this.logger.error("A valid schema couldn't be composed.")
-      } else {
-       this.queryPlannerPointer = getQueryPlanner(composedSdl);
-      }
+    // And also support the `debug` option, if it's truthy.
+    if (this.config.debug === true) {
+      loglevelLogger.setLevel(loglevelLogger.levels.DEBUG);
+    } else {
+      loglevelLogger.setLevel(loglevelLogger.levels.WARN);
     }
 
-    if (isCsdlConfig(this.config)) {
-      const { schema } = this.createSchema({ csdl: this.config.csdl });
-      this.schema = schema;
-      this.queryPlannerPointer = getQueryPlanner(this.config.csdl);
-    }
+    return loglevelLogger;
+  }
 
-    this.initializeQueryPlanStore();
-
-    // this will be overwritten if the config provides experimental_updateServiceDefinitions
-    this.updateServiceDefinitions = this.loadServiceDefinitions;
-
-    if (config) {
-      this.updateServiceDefinitions =
-        config.experimental_updateServiceDefinitions ||
-        this.updateServiceDefinitions;
-      // set up experimental observability callbacks
-      this.experimental_didResolveQueryPlan =
-        config.experimental_didResolveQueryPlan;
-      this.experimental_didFailComposition =
-        config.experimental_didFailComposition;
-      this.experimental_didUpdateComposition =
-        config.experimental_didUpdateComposition;
-
-      this.experimental_approximateQueryPlanStoreMiB =
-        config.experimental_approximateQueryPlanStoreMiB;
-
-      if (
-        isManagedConfig(config) &&
-        config.experimental_pollInterval &&
-        config.experimental_pollInterval < 10000
-      ) {
-        this.experimental_pollInterval = 10000;
-        this.logger.warn(
-          'Polling Apollo services at a frequency of less than once per 10 seconds (10000) is disallowed. Instead, the minimum allowed pollInterval of 10000 will be used. Please reconfigure your experimental_pollInterval accordingly. If this is problematic for your team, please contact support.',
-        );
-      } else {
-        this.experimental_pollInterval = config.experimental_pollInterval;
-      }
+  private initQueryPlanStore(size?: number) {
```

Comment:
maybe nice for the arg to have units, though it is purely private. `approximateQueryPlanStoreMiB` would be a fine name?

Reply:
de97278